File: frontend/types/document.ts
export interface Document {
    DocumentID: number;
    FileName: string;
    OriginalName: string;
    FileType: string;
    FileSize: number;
    FilePath: string;
    UploadedBy: number;
    CreatedAt: string;
  }
  
  export interface DocumentCreate {
    FileName: string;
    OriginalName: string;
    FileType: string;
    FileSize: number;
    FilePath: string;
    UploadedBy: number;
  }
  
  export interface DocumentUpdate {
    FileName: string;
    OriginalName: string;
    FileType: string;
    FileSize: number;
    FilePath: string;
    UploadedBy: number;
  }

File: backend/app/schemas/document.py
from pydantic import BaseModel
from datetime import datetime


class DocumentBase(BaseModel):
    FileName: str
    OriginalName: str
    FileType: str
    FileSize: int
    FilePath: str
    UploadedBy: int


class DocumentCreate(DocumentBase):
    pass


class DocumentUpdate(DocumentBase):
    pass


class DocumentResponse(DocumentBase):
    DocumentID: int
    CreatedAt: datetime

    class Config:
        from_attributes = True